checkov/common/util/http_utils.py

```diff
@@ -25,11 +25,27 @@
 logger = logging.getLogger(__name__)
 
 
+@overload
+def normalize_bc_url(url: None) -> None:
+    ...
+
+
+@overload
+def normalize_bc_url(url: str) -> str:
+    ...
+
```

Comment:
HI! I 
never saw such a using of overload.
can you please share why those 2 function are necessary?
it seem that the third one is enough

Reply:
the `@overload` functionality in Python is only for typing purposes. It gives a possibility to differentiate between what kind of types go in and what to what should come out. In this example it is clear, if `None` goes in, `None` should come out, same for `str`.
If you don't add the `@overload` annotation, then the type checker will assume that either `str` or `None` could return, which can be ok, but then you have handle it downstream and always check, if it `is not None`

Reply:
cool!